nats.go

```diff
@@ -2448,17 +2455,29 @@ func NewInbox() string {
 	return string(b[:])
 }
 
-// Creates a new literal response subject that will trigger
-// the global subscription handler.
+// newRespInbox creates a new literal response subject
+// that will trigger the mux subscription handler.
+// Lock should be held.
 func (nc *Conn) newRespInbox() string {
-	var b [inboxPrefixLen + (2 * nuidSize) + 1]byte
+	var b [respInboxPrefixLen + replySuffixLen]byte
 	pres := b[:respInboxPrefixLen]
 	copy(pres, nc.respSub)
-	ns := b[respInboxPrefixLen:]
-	copy(ns, nuid.Next())
+	rn := nc.respRand.Int63()
+	for i, l := respInboxPrefixLen, rn; i < len(b); i++ {
+		b[i] = rdigits[l%base]
+		l /= base
+	}
 	return string(b[:])
 }
 
+// NewRespInbox is the new format used for _INBOX.
+func (nc *Conn) NewRespInbox() string {
+	nc.mu.Lock()
+	s := nc.newRespInbox()
```

Comment:
`nc.NewRespInbox()` will fail if it is attempted to be used without having called `Request` before, maybe should check here and initialize if undefined?

```diff
diff --git a/nats.go b/nats.go
index aaf4995..bb92cdc 100644
--- a/nats.go
+++ b/nats.go
@@ -2473,6 +2473,11 @@ func (nc *Conn) newRespInbox() string {
 // NewRespInbox is the new format used for _INBOX.
 func (nc *Conn) NewRespInbox() string {
 	nc.mu.Lock()
+	if nc.respMap == nil {
+		nc.respSub = fmt.Sprintf("%s.*", NewInbox())
+		nc.respMap = make(map[string]chan *Msg)
+		nc.respRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+	}
 	s := nc.newRespInbox()
 	nc.mu.Unlock()
 	return s
```

Reply:
Good point, will fix.